### ISSUE TITLE
Gradient Descent

### ISSUE THREAD

**Issue by johncava** [johncava/NN_XOR#1]
Need to implement the algorithm for using gradient descent to train the neural network
